project/make/package-rpm

```diff
@@ -0,0 +1,136 @@
+#!/bin/bash
+
+DEST=$1
+
+PKGVERSION="${VERSION//-/'~'}"
+# if we have a "-dev" suffix or have change in Git, let's make this package version more complex so it works better
+if [[ "$VERSION" == *-dev ]] || [ -n "$(git status --porcelain)" ]; then
+	GIT_UNIX="$(git log -1 --pretty='%at')"
+	GIT_DATE="$(date --date "@$GIT_UNIX" +'%Y%m%d.%H%M%S')"
+	GIT_COMMIT="$(git log -1 --pretty='%h')"
+	GIT_VERSION="git${GIT_DATE}.0.${GIT_COMMIT}"
+	# GIT_VERSION is now something like 'git20150128.112847.0.17e840a'
+	PKGVERSION="$PKGVERSION~$GIT_VERSION"
+fi
+
+# probably shouldn't be using dpkg here...
+PACKAGE_ARCHITECTURE="$(dpkg-architecture -qDEB_HOST_ARCH)"
```

Comment:
Looks like the author of fpm handles arch translations appropriately (you can pass deb-form arch, and when target is rpm, it will convert properly.  (https://github.com/jordansissel/fpm/blob/master/lib/fpm/package/rpm.rb#L205-L217) I agree that `uname -m` will be best here.